### PR TITLE
Raven: update description, test url, and routing; fix tests

### PR DESCRIPTION
**Pull request by bslobodin**
@girasquid @bizla this fixes remote integration tests for `pac_net_raven` and updates description to be less about "PacNet" since other entities out there use this as their gateway.

:bow: 

```
➜  active_merchant git:(fix_raven_remote_tests) bundle exec rake test:units TEST=test/unit/gateways/pac_net_raven_test.rb
Started
..........................................
Finished in 0.018617 seconds.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
42 tests, 132 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
2256.00 tests/s, 7090.29 assertions/s

➜  active_merchant git:(fix_raven_remote_tests) bundle exec rake test:remote TEST=test/remote/gateways/remote_pac_net_raven_test.rb
Started
................
Finished in 26.262058 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
16 tests, 132 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.61 tests/s, 5.03 assertions/s
```

### CODE REVIEW

**Comment by bslobodin** [comment]
I butchered original comment, re-opening.

File: test/remote/gateways/remote_pac_net_raven_test.rb [hunk 1]
```diff
@@ -50,10 +50,10 @@ def test_expired_credit_card_purchese
     assert_equal "Invalid because the card expiry date (mmyy) \"0912\" is not a date in the future", purchase.params['Message']
   end
 
-  def test_declined_purchese
+  def test_declined_purchase
```
Comment (bizla) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: :heart_eyes:

File: lib/active_merchant/billing/gateways/pac_net_raven.rb [hunk 2]
```diff
@@ -25,15 +25,15 @@ class PacNetRavenGateway < Gateway
         'cvv2_not_checked' => 'X'
       }
 
-      self.test_url = 'https://demo.deepcovelabs.com/realtime/'
-      self.live_url = 'https://raven.pacnetservices.com/realtime/'
+      self.test_url = 'https://raven.deepcovelabs.com/realtime/'
```
Comment (bizla) [inline lib/active_merchant/billing/gateways/pac_net_raven.rb]: No test endpoint anymore?

File: test/remote/gateways/remote_pac_net_raven_test.rb [hunk 3]
```diff
@@ -156,15 +156,15 @@ def test_authorize_and_void
   def test_authorize_capture_and_void
     auth = @gateway.authorize(@amount, @credit_card, @options)
     assert capture = @gateway.capture(@amount, auth.authorization)
-    assert void = @gateway.void(capture.authorization, {:pymt_type =>  capture.params['PymtType']})
-    assert_failure void
+    assert void = @gateway.void(capture.authorization, {:pymt_type => capture.params['PymtType']})
+    assert_success void
```
Comment (bizla) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: Old endpoint didn't like voids?

File: lib/active_merchant/billing/gateways/pac_net_raven.rb (hunk 2 above)
Comment (bslobodin) [inline lib/active_merchant/billing/gateways/pac_net_raven.rb]: Correct. Instead, we set up different routings for folks to run their tests against on the production system.

**Comment by bizla** [comment]
Just some business questions, code LGTM.

File: test/remote/gateways/remote_pac_net_raven_test.rb (hunk 3 above)
Comment (bslobodin) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: That was going to be my guess as well, and whomever wrote the original tests just asserted whatever results they were seeing, but really, voiding results of the capture is perfectly valid, until it's settled overnight that is. For the purposes of this test, there's no reason for void to fail right away.

File: lib/active_merchant/billing/gateways/pac_net_raven.rb (hunk 2 above)
Comment (bizla) [inline lib/active_merchant/billing/gateways/pac_net_raven.rb]: :cool: :sunglasses:

File: test/remote/gateways/remote_pac_net_raven_test.rb (hunk 3 above)
Comment (bizla) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: :+1:

File: test/remote/gateways/remote_pac_net_raven_test.rb [hunk 4]
```diff
@@ -156,15 +156,15 @@ def test_authorize_and_void
   def test_authorize_capture_and_void
     auth = @gateway.authorize(@amount, @credit_card, @options)
     assert capture = @gateway.capture(@amount, auth.authorization)
-    assert void = @gateway.void(capture.authorization, {:pymt_type =>  capture.params['PymtType']})
-    assert_failure void
+    assert void = @gateway.void(capture.authorization, {:pymt_type => capture.params['PymtType']})
```
Comment (girasquid) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: What happens if you void without `pymt_type`?

**Comment by girasquid** [comment]
One question (in a comment), and also wondering: can you do partial captures? I only see a test for the overcapture on an auth failing and now I'm curious.

File: test/remote/gateways/remote_pac_net_raven_test.rb (hunk 4 above)
Comment (bslobodin) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: If you try voiding without proving a matching payment type, the system won't find the original payment to void ¯\_(ツ)_/¯

Comment (girasquid) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: Can you tweak the integration to return an authorization that includes the authorization code + the payment type, if that's required to be able to void? The expectation is that calling `gateway.void(authorization)` is all you need to void something.

The Authorize.Net adapter does this by joining things together with `#`: https://github.com/activemerchant/active_merchant/blob/93251aaf947361f3cab3be4d2356ecc636d928c7/lib/active_merchant/billing/gateways/authorize_net.rb#L736-L743

**Comment by bslobodin** [comment]
@girasquid Good question, it used to be allowed, but now we only allow an exact same amount to be captured; there were too many scheme rules around what's allowed when and we went the simplest route given our merchants' needs. Partial refunds are, of course, still allowed, up to and including the original amount; no over-refunding.

File: test/remote/gateways/remote_pac_net_raven_test.rb (hunk 4 above)
Comment (bslobodin) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: Good point, leave it with me.

Comment (girasquid) [inline test/remote/gateways/remote_pac_net_raven_test.rb]: :heart_eyes:

**Comment by girasquid** [comment]
Makes sense - thanks for the clarification! Ping when you're ready for re-review.

**Comment by bslobodin** [comment]
@girasquid the issue wasn't so much on our end, just `void` deciding to default to `cc_debit`, limiting the lookup here. Should be better now.

```
➜  active_merchant git:(fix_raven_remote_tests) bundle exec rake test:remote TEST=test/remote/gateways/remote_pac_net_raven_test.rb
Started
................
Finished in 20.945151 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
16 tests, 132 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.76 tests/s, 6.30 assertions/s
```

**Comment by bizla** [comment]
:pizza:

**Comment by girasquid** [comment]
Awesome - looks good. Thanks!

**Pull request closed without merging**